File: OneQuestionPerDay/1021/remove_outermost_parentheses.go
package main

import "strings"

func removeOuterParentheses(S string) string {
	var strs []string
	start := 0
	counter := 0
	for i, c := range S {
		if c == '(' {
			counter++
		} else {
			counter--
		}
		if counter == 0 {
			strs = append(strs, S[start+1:i])
			start = i + 1
		}
	}
	return strings.Join(strs, "")
}

File: OneQuestionPerDay/1021/remove_outermost_parentheses.py
class Solution:
    def removeOuterParentheses(self, S: str) -> str:
        counter = 0
        start = 0

        strs = []
        for i, c in enumerate(S):
            if c == "(":
                counter += 1
            else:
                counter -= 1

            if counter == 0:
                strs.append(S[start + 1:i])
                start = i + 1
        return "".join(strs)
